signer/src/eth.rs

```diff
@@ -278,10 +150,99 @@ pub mod dev {
 
 #[cfg(feature = "subxt")]
 mod subxt_compat {
+    use super::*;
+    use crate::ecdsa;
     use subxt_core::config::Config;
     use subxt_core::tx::signer::Signer as SignerT;
+    use subxt_core::utils::AccountId20;
+    use subxt_core::utils::MultiAddress;
+
+    impl Keypair {
```

Comment:
Ah this isn't quite what I had in mind; I was hitnking you could do something more like sr25519.rs and have eg:

```rust
/// The public key for an [`Keypair`] key pair. This is equivalent to an `AccountId20`.
pub struct PublicKey(pub [u8; 20]);

pub struct Keypair(..)

// Impl here; not gated by subxt-compat feature:
impl Keypair {
    // ...
}

#[cfg(feature = "subxt")]
mod subxt_compat {
    use super::*;

    // Only in this block do we now need to import subxt-core to
    // describe how to convert our PublicKey above into an AccountId20:
    use subxt_core::utils::AccountId20;

    impl From<PublicKey> for AccountId20 {
        fn from(value: PublicKey) -> Self {
            value.to_account_id()
        }
    }

    impl PublicKey {
        /// A shortcut to obtain an [`AccountId20`] from a [`PublicKey`].
        /// We often want this type, and using this method avoids any
        /// ambiguous type resolution issues.
        pub fn to_account_id(self) -> AccountId20 {
            AccountId20(self.0)
        }
    }

    impl<T: Config> SignerT<T> for Keypair
    where
        T::AccountId: From<PublicKey>,
        // looks like T::Address is also AcocuntId20 so the one From impl above works,
        // but this allows other address types to be used as long as there's a way to convert:
        T::Address: From<PublicKey>, 
        // sameish: we just use the Signature type above so I think this Just Works, but
        // also leaves door open for people to use other Signature types in their Config as 
        // long as they provide a From impl:
        T::Signature: From<Signature>,
    {
        fn account_id(&self) -> T::AccountId {
            self.public_key().into()
        }

        fn address(&self) -> T::Address {
            self.public_key().into()
        }

        fn sign(&self, signer_payload: &[u8]) -> T::Signature {
            self.sign(signer_payload).into()
        }
    }
}
```

Reply:
updated in line with the suggestion